packages/block-editor/src/hooks/effects.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { hasBlockSupport } from '@wordpress/blocks';
+import { useSelect } from '@wordpress/data';
+/**
+ * Internal dependencies
+ */
+import StylesEffectsPanel, {
+	useHasEffectsPanel,
+} from '../components/global-styles/effects-panel';
+import { InspectorControls } from '../components';
+import { store as blockEditorStore } from '../store';
+
+export const EFFECTS_SUPPORT_KEYS = [ 'shadow' ];
+
+export function hasEffectsSupport( blockName ) {
+	return EFFECTS_SUPPORT_KEYS.some( ( key ) =>
+		hasBlockSupport( blockName, key )
+	);
+}
+
+function EffectsInspectorControl( { children, resetAllFilter } ) {
+	return (
+		<InspectorControls group="effects" resetAllFilter={ resetAllFilter }>
+			{ children }
+		</InspectorControls>
+	);
+}
+
+export function EffectsPanel( { clientId, setAttributes, settings } ) {
+	const isEnabled = useHasEffectsPanel( settings );
+	const onChange = ( newShadow ) => {
+		setAttributes( styleToAttributes( newShadow ) );
+	};
+	const blockAttributes = useSelect(
+		( select ) => select( blockEditorStore ).getBlockAttributes( clientId ),
+		[ clientId ]
+	);
+	const shadow = blockAttributes?.shadow;
```

Comment:
At the block level, would it be better to keep the `shadow` prop within the `style` object? I think most block supports have been leaning toward keeping everything in that object where possible.

Reply:
Thanks for that clarification @madhusudhand 👍 

In that case, my vote would be for the shadow attribute to be moved within the style object. 

As for the class names, I don't hold any strong opinions. I take it the class names are being added due to the use of presets? If pushed I would lean towards still adding the class names but very happy to leave that decision to others that know more about shadow styles 🙂